TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -93,73 +117,110 @@ inline void step(const VMStubMECM<VMSMEType> stubmem[4][1024]) {
   //If the buffer is not empty we have a projection that we need to 
   //process. 
   
-  NSTUBS istubtmp=istub_;
+  NSTUB istubtmp=istub_;
+  NSTUB iusetmp=iuse_;
 
-  ap_uint<VMStubMECMBase<VMSMEType>::kVMSMEFinePhiSize> iphiSave = iphi_ + phiPlus_;
+  ap_uint<VMStubMECMBase<VMSMEType>::kVMSMEFinePhiSize> iphiSave = iphi_ + use_[iusetmp].range(0,0);
   auto secondSave = second_;
 
+  VMProjection<VMProjType> data(projbuffer_.getProjection());
+  bool useSecond = data.getZBin().range(0,0)==1;
+  constexpr bool isDisk = LAYER > TF::L6;
+  constexpr int nbins = isDisk ? (1 << kNbitsrzbin)*2 : (1 << kNbitsrzbin); //twice as many bins in disks (since there are two disks)
+  constexpr regionType APTYPE = TF::layerDiskRegion[LAYER];
+  int sign = isDisk ? (projbuffer_.getPhiDer() < 0 ? -1 : 1) : 0;
+  bool usefirstMinus = use_[iusetmp] == 0;
+  bool usesecondMinus = use_[iusetmp] == 2;
+  bool usefirstPlus = use_[iusetmp] == 1;
+  bool usesecondPlus = use_[iusetmp] == 3;
+  bool first = usefirstPlus || usefirstMinus;
+  bool second = usefirstPlus || usesecondPlus;
```

Comment:
Is this correct? Should this not be bool second  = usesecontplzs || useSecondMinus? (If the code is correct then the variable name is very confusing.)

Reply:
I no longer use `first` and `second`, so I removed them.